### PR TITLE
[PW-7805] Timeout variable for HttpClient

### PR DESCRIPTION
**Pull request by jillingk**
**Description**
This PR adds a timeout variable in the config, that sets the timeout for the client constructors with a default httpclient instance. 

**Tested scenarios**
Since the HttpClient object is private readonly in the HttpClientWrapper object, it's hard to test with UT. However I verified with a debugger that the timeout is correctly set.

### CODE REVIEW

File: Adyen/Client.cs [hunk 1]
```diff
@@ -53,7 +54,12 @@ public Client(string username, string password, Environment environment, string
                 Environment = environment
             };
             SetEnvironment(environment, liveEndpointUrlPrefix);
-            _client = new HttpClientWrapper(Config, new System.Net.Http.HttpClient());
+            
+            // Set Timeout for HttpClient
+            var httpClient = new System.Net.Http.HttpClient();
+            httpClient.Timeout = Config.Timeout == default ? TimeSpan.FromMilliseconds(100000) : TimeSpan.FromMilliseconds(Config.Timeout);
```
Comment (michaelpaul) [inline Adyen/Client.cs:60]: Let's assign a timeout only if the user explicitly provides one. 
```suggestion
            if (Config.Timeout != default)
            {
                httpClient.Timeout = TimeSpan.FromMilliseconds(Config.Timeout);
            }

```

**Review by michaelpaul — CHANGES_REQUESTED** [review]

File: Adyen/Client.cs (hunk 1 above)
Comment (jillingk) [reply Adyen/Client.cs:60]: Really wanted to use a one liner here and the default is 100000 milliseconds haha

File: Adyen/Client.cs [hunk 2]
```diff
@@ -66,14 +75,30 @@ public Client(string xapikey, Environment environment, string liveEndpointUrlPre
...
+            _client = new HttpClientWrapper(Config, httpClient);
         }
 
         public Client(Config config)
         {
             Config = config;
             SetEnvironment(Config.Environment, Config.LiveEndpointUrlPrefix);
-            _client = new HttpClientWrapper(Config, new System.Net.Http.HttpClient());
+            
+            // Set Timeout for HttpClient
```
Comment (wboereboom) [inline Adyen/Client.cs:94]: can we move this code to a helper function rather than copying it three times?

Comment (jillingk) [reply Adyen/Client.cs]: Done

**Review by michaelpaul — APPROVED** [review]

**Review by AlexandrosMor — APPROVED** [review]

**Pull request merged**